actions/platform/service_account/update_api_integration.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any, Text, Dict, List
+
+from rasa_sdk import Action, Tracker, FormValidationAction
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.events import SlotSet, ActiveLoop
+from rasa_sdk.types import DomainDict
+
+from common import logging
+from common.config import app
+from actions.slot_match import FuzzySlotMatch, FuzzySlotMatchOption, resolve_slot_match
+
+logger = logging.initialize_logging()
+
+DOCS_OR_ACCOUNT_SLOT = "docs_or_walkthrough"
+HAS_SERVICE_ACC_CREDS = "has_service_account_creds"
+NEED_ANOTHER_INTEGRATION = "need_another_integration"
+
+docs_or_account_choice = FuzzySlotMatch(
+    "docs_or_walkthrough",
+    [
+        FuzzySlotMatchOption("docs", ["docs", "documentation", "doc"]),
+        FuzzySlotMatchOption(
+            "walkthrough",
+            ["walkthrough", "walk"],
+        ),
+    ],
+)
+
+
+class UpdateApiIntegration(FormValidationAction):
+    def name(self) -> Text:
+        return "validate_form_update_api_integration"
+
+    @staticmethod
+    def extract_docs_or_walkthrough(
+        dispatcher: CollectingDispatcher, tracker: Tracker, domain: DomainDict
+    ) -> Dict[Text, Any]:
+        if tracker.get_slot("requested_slot") == DOCS_OR_ACCOUNT_SLOT:
+            resolved = resolve_slot_match(
+                tracker.latest_message["text"], docs_or_account_choice, accepted_rate=95
+            )
+
+            if len(resolved) > 0:
+                return resolved
+
+            return {DOCS_OR_ACCOUNT_SLOT: None}
+
+        return {}
+
+    @staticmethod
+    def validate_docs_or_walkthrough(
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: DomainDict,
+    ) -> Dict[Text, Any]:
+        return {DOCS_OR_ACCOUNT_SLOT: value}
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[Dict[Text, Any]]:
+        requested_slot = tracker.get_slot("requested_slot")
+
+        if tracker.get_slot(DOCS_OR_ACCOUNT_SLOT) == "docs":
+            dispatcher.utter_message(response="utter_docs_only_redirect")
+            return [
+                SlotSet(HAS_SERVICE_ACC_CREDS, False),
+                SlotSet(NEED_ANOTHER_INTEGRATION, False),
+            ]
+
+        if tracker.get_slot(DOCS_OR_ACCOUNT_SLOT) == "walkthrough":
+            if requested_slot == HAS_SERVICE_ACC_CREDS:
+                if tracker.get_slot(HAS_SERVICE_ACC_CREDS) == True:
+                    dispatcher.utter_message(
```

Comment:
Holy mackerel, that's a lot of messages! How does it look in the window?

Reply:
I am not able to get a screenshot at the moment because the local frontend seems to be broken.
If I recall correctly, I think it didn't seem to bad. The markdown formats kind of separates the big checks decently.
Merging it now, and we can check it out in stage. :+1:

Reply:
Looks kind of okay, I think:
![image](https://github.com/RedHatInsights/astro-virtual-assistant/assets/17099954/e734ef70-787c-4d11-9116-4efd768c5e94)

Reply:
The window isn't usually that big though :thinking: